src/logic/basic.lean

```diff
@@ -1144,6 +1144,12 @@ hpq _ $ some_spec _
 noncomputable def subtype_of_exists {α : Type*} {P : α → Prop} (h : ∃ x, P x) : {x // P x} :=
 ⟨classical.some h, classical.some_spec h⟩
 
+protected noncomputable def peirce {α : Sort*} (H : (α → false) → false) : α :=
+classical.choice $ peirce _ false $ λ h, (H $ λ a, h ⟨a⟩).elim
```

Comment:
Should this have a docstring referencing https://en.wikipedia.org/wiki/Peirce%27s_law?

Reply:
On review, this is actually closer to double negation elimination aka `by_contradiction`, rather than Peirce's law. I've changed the names.